### ISSUE TITLE
Generate default bundle config from configschema.json

### ISSUE THREAD

**Issue by denolfe** [nodecg/nodecg-cli#12]
I looked into this last night after speaking briefly with @Lange on gitter. Generating questions for inquirer.js to question the user, then populate a config is not feasible due to how complex a schema can be.

However, generating a config based on the `configschema.json` can be done rather easily using something like [JSON-Schema-Instantiator](https://github.com/tomarad/JSON-Schema-Instantiator), which supports the `"default"` tag in JSON Schema.

For instance:

``` json
{
  "properties": {
    "twitchUsername": {
      "type": "string"
    },
    "updateViewerInterval": {
      "description": "Interval to update viewer count (seconds)",
      "type": "integer",
      "default": 15,
      "minimum": 5
    },
    "updateFollowerInterval": {
      "description": "Interval to update followers (seconds)",
      "type": "integer",
      "default": 15,
      "minimum": 5
    }
  },
  "required": ["twitchUsername"]
}
```

would output

``` json
{
  "twitchUsername": "",
  "updateViewerInterval": 15,
  "updateFollowerInterval": 15
}
```

Is this something that would still be useful?

**Comment by Lange** [comment]
Thanks for taking the time to look into this @denolfe!

The NodeCG core uses [`json-schema-defaults`](https://www.npmjs.com/package/json-schema-defaults) to populate default values. For consistency's sake, I'd like to use that here as well. That said, if there's some killer feature of [`json-schema-instantiator`](https://github.com/tomarad/JSON-Schema-Instantiator) that I'm not aware of, I can be swayed.

I'm fully on-board with this idea of generating a default config for a bundle based on its `configschema.json`. The only question I have is what will this command be called? :smile: 

I'm unsure what a concise, short name for this command would be.

**Comment by Lange** [comment]
Also, it probably makes sense to automatically run this command when installing a bundle for the first time.

**Comment by denolfe** [comment]
I'll look into using `json-schema-defaults`.

The name.... Yeah, I'm unsure as well. `nodecg defaultconfig` is ugly, but `nodecg config` is a bit vague.

**Comment by Lange** [comment]
I agree that `nodecg config` is too vague and not suitable. Perhaps `nodecg config default`? Though that's odd, because `nodecg config` doesn't do anything, and there's no plans for it to do anything. Maybe `nodecg defaultconfig` is the way to go.

If we can't think of anything better, I'm happy to go with `nodecg defaultconfig`.

**Comment by denolfe** [comment]
Closed by e247110
